Client/Frontend/Browser/SearchLoader.swift

```diff
@@ -86,7 +115,12 @@ class SearchLoader: Loader<Cursor<Site>, SearchViewController>, FeatureFlaggable
                     GleanMetrics.Awesomebar.queryTime.stopAndAccumulate(timerid)
                 }
 
-                guard !deferredHistory.cancelled else { return }
+                let cancellableHistory = deferredHistory as? CancellableDeferred
+                if let cancellableHistory = cancellableHistory {
+                    if cancellableHistory.cancelled {
```

Comment:
Can be done in one `if`, a bit easier to read IMO
```suggestion
                if let cancellableHistory = cancellableHistory, cancellableHistory.cancelled {
```